analyses/snv-callers/compare_snv_callers_plots-tcga.Rmd

```diff
@@ -0,0 +1,397 @@
+- [VAF correlations between callers](#vaf-correlations)
+
+#### Usage
+
+To run this from the command line, use:
+```
+Rscript -e "rmarkdown::render('analyses/snv-callers/compare_snv_callers_plots-tcga.Rmd', 
+                              clean = TRUE)"
+```
+
+_This assumes you are in the top directory of the repository._
+
+## Setup
+
+#### Packages and functions
+
+Read in set up script.
+
+```{r}
+if (!("ggupset" %in% installed.packages())) {
+  install.packages("ggupset", repos = "http://cran.us.r-project.org")
+}
+if (!("UpSetR" %in% installed.packages())) {
+  install.packages("UpSetR", repos = "http://cran.us.r-project.org")
+}
+if (!("GGally" %in% installed.packages())) {
+  install.packages("GGally", repos = "http://cran.us.r-project.org")
+}
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+```
+
+Set up directories. 
+
+```{r}
+scratch_dir <- file.path("..", "..", "scratch")
+results_dir <- file.path("results", "tcga-comparison")
+plots_dir <- file.path("plots", "tcga-comparison")
+
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+```
+
+```{r}
+upset_plot <- function(callers_per_mutation_df) {
+callers_per_mutation_df %>% 
+  dplyr::group_by(caller_combo) %>%
+  dplyr::tally() %>% 
+  ggplot2::ggplot(ggplot2::aes(x = reorder(caller_combo, -n), y = n)) +
+  ggplot2::geom_bar(
+        position = "dodge", stat = "identity") +
+  ggplot2::theme_classic() +
+  ggupset::scale_x_mergelist(sep = "-") +
+  ggupset::axis_combmatrix(sep = "-") +
+  ggplot2::xlab("") +
+  ggplot2::ylab("")
+}
+```
+
+## Connect to TCGA database
+
+Connect to SQLite database.
+
+```{r}
+# Start up connection
+con <- DBI::dbConnect(RSQLite::SQLite(), 
+                      file.path(scratch_dir, "tcga_snv_db.sqlite"))
+```
+
+Note what columns we will join by.
+
+```{r}
+join_cols = c("Chromosome",
+              "Start_Position",
+              "Reference_Allele",
+              "Allele",
+              "Tumor_Sample_Barcode", 
+              "Variant_Classification")
+```
+
+Set up tables from the database but only call the columns we need.
+
+```{r}
+strelka <- dplyr::tbl(con, "strelka") %>% 
+  dplyr::select(join_cols, "VAF") %>% 
+  as.data.frame()
+
+lancet <- dplyr::tbl(con, "lancet") %>% 
+  dplyr::select(join_cols, "VAF") %>% 
+  as.data.frame()
+
+mutect <- dplyr::tbl(con, "mutect") %>% 
+  dplyr::select(join_cols, "VAF") %>% 
+  as.data.frame()
+```
+
+Because `DBI` does not support full_join, we had to use a series of `left_join` and
+`union_all` calls in order to get a full join of all the callers. 
+
+```{r}
+# Bring out the data.frame
+all_caller_df <- strelka %>%
+  dplyr::full_join(mutect, by = join_cols, 
+                   suffix = c("_strelka", "_mutect")) %>%
+  dplyr::full_join(lancet, 
+                   by = join_cols) %>% 
+  dplyr::rename(VAF_lancet = VAF)
+
+# Take a peek at what this looks like
+head(all_caller_df)
+```
+
+```{r}
+# Set up the data as a data.frame
```

Comment:
This comment does not seem to reflect the lines below. I think it would be worth writing a bit more in the text before this code chunk to explain what these steps are doing.